### DISCUSSION TITLE
API authorizatie werkt niet?

### DISCUSSION THREAD

**Discussion by n9iels** [dsmrreader/dsmr-reader#1519] [Questions & answers / How-to's] [answered]
### Description

Ik probeer de DSMR-Reader API te aan te gebruiken, maar wat ik ook doe ik blijft een HTTP 403 terug krijgen. Ik roep bijv. het endpoint `/api/v2/application/monitoring` met de volgende header `Authorization Token xxx`. Andere endpoints zoals ` `/api/v2/application/status` doen het ook niet.

Ik snap er niks van. De API key is juist, want deze wordt ook actief gebruik door mijn remote-datalogger en het script daarachter gebruikt op dezelfde manier de Authorization header. Ik snap er niks van 😅 Wat zie ik over het hoofd?

### DSMR-reader version

4.19

### DSMR-reader platform

Docker (e.g. Xirixiz)

**Comment by dennissiemensma** [comment]
Wat krijg je met bijvoorbeeld CURL?

```
curl 'http://localhost/api/v2/datalogger/dsmrreading?ordering=-timestamp&limit=1' -H 'Authorization: Token Test'

{"detail":"Invalid auth key"}
```

```
curl 'http://localhost/api/v2/datalogger/dsmrreading?ordering=-timestamp&limit=1' -H 'Authorization: Token 8GZ...'

{"count":132632,"next":"http://localhost/api...
```

**Comment by n9iels** [comment, marked as answer]
Ik heb in de gaten wat er aan de hand is, en dat ligt niet aan jou software 😓 🤦 Op de server waar DSMR-reader staat draait een remote-proxy die de Authorization header net op een andere manier forward... Ik meot `X-Api-Token` als header gebruiken, deze wordt dan intern omgezet naar een `Authorization` header. Als ik dat gebruikt werkt het wel geweoon netjes.

Excuus en bedankt voor de hulp!

**Comment by dennissiemensma** [comment]
Bedankt voor je aanvulling, fijn dat je het zelf hebt kunnen vinden. Dit soort zaken blijven soms lastig te debuggen hoor!
